### PR TITLE
As/shared components

### PR DESCRIPTION
**Pull request by AzizSandid**
## What does this do?

decompose checkbox file into three files

## Why did you do this?

To facilitate the use of the component checkbox, radio, and switch.

### CODE REVIEW

File: src/shared/components/radio.tsx [hunk 1]
```diff
@@ -0,0 +1,82 @@
+import { MotiView } from 'moti';
+import React from 'react';
+
+import colors from '@/theme/colors';
+
+import type { IconProps, RootProps } from './checkbox';
+import { Root } from './checkbox';
+import { Text } from './text';
+
+const SIZE = 20;
```
Comment (MohamedMegdich1) [inline src/shared/components/radio.tsx:10]: Please avoid setting static numbers for the size.

File: src/shared/components/switch.tsx [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+import React from 'react';
+import { I18nManager, View } from 'react-native';
+
+import colors from '@/theme/colors';
+
+import type { IconProps, RootProps } from './checkbox';
+import { Root } from './checkbox';
+import { Text } from './text';
+
+const WIDTH = 50;
```
Comment (MohamedMegdich1) [inline src/shared/components/switch.tsx:11]: the same

File: src/shared/components/radio.tsx [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+  const color = checked ? colors.primary[300] : colors.charcoal[400];
+  return (
+    <MotiView
+      style={{
+        height: SIZE,
+        width: SIZE,
+        borderColor: color,
+      }}
+      className="items-center justify-center rounded-[20px] border-2 bg-transparent"
+      from={{ borderColor: '#CCCFD6' }}
```
Comment (MohamedMegdich1) [inline src/shared/components/radio.tsx:36]: Remember: Do not use a static color use theme instead.

**Review by MohamedMegdich1 — CHANGES_REQUESTED** [review]

File: src/shared/components/switch.tsx [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+      <View className="overflow-hidden rounded-full">
+        <View
+          style={{
+            backgroundColor,
+          }}
+          className="h-8 w-16"
+        />
+      </View>
+      <MotiView
+        // eslint-disable-next-line react-native/no-inline-styles
```
Comment (khalil7898) [inline src/shared/components/switch.tsx:41]: why not fixing the real issue instead of disabling the linter?

File: src/theme/colors.js [hunk 5]
```diff
@@ -75,4 +75,12 @@ module.exports = {
     800: '#991B1B',
     900: '#7F1D1D',
   },
+  bgm: {
+    //background colors for progress bar and second background
```
Comment (khalil7898) [inline src/theme/colors.js:79]: do we need to move this to the theme and check each them rendering?

File: src/shared/components/switch.tsx (hunk 4 above)
Comment (AzizSandid) [reply src/shared/components/switch.tsx:41]: It wasn't me who disabled it. We installed it from Obytes like that.

File: src/theme/colors.js (hunk 5 above)
Comment (AzizSandid) [reply src/theme/colors.js:79]: i will fix it

**Pull request merged**